crates/red_knot_python_semantic/src/types.rs

```diff
@@ -1638,6 +2005,30 @@ impl<'db> Type<'db> {
                         };
                     }
 
+                    Some(KnownFunction::GetattrStatic) => {
+                        let Some((instance_ty, attr_name, default)) =
+                            binding.three_parameter_types()
+                        else {
+                            return Ok(CallOutcome::Single(binding));
+                        };
+
+                        let Some(attr_name) = attr_name.into_string_literal() else {
+                            return Ok(CallOutcome::Single(binding));
+                        };
+
+                        let default = if default.is_unknown() {
+                            Type::Never
+                        } else {
+                            default
+                        };
+
+                        let static_member = instance_ty
+                            .static_member(db, attr_name.value(db))
+                            .ignore_possibly_unbound() // TODO: we could emit a diagnostic here (if default is not set)
+                            .unwrap_or(default);
+                        binding.set_return_type(static_member);
+                    }
```

Comment:
I think this gives us less-than-ideal behaviour for something like this:

```py
from inspect import getattr_static
from typing import Any, reveal_type

def _(x: Any):
    reveal_type(getattr_static(x, "foo", 42))
```

We reveal `Any` here on this branch, but a more precise type would be `Any | Literal[42]`

Reply:
I suppose this would probably require some maybe-ugly special casing for dynamic types...

Reply:
> I don't think so. `Any` is the correct type for this example. If the attribute `foo` actually exists, it could literally be of _any_ type:

Right... but if it doesn't exist, the returned value of `getattr_static(x, "foo", 42)` will be `42`. By including `Literal[42]` in the union, we force the user to make sure that any operations they attempt on the returned value are valid for values of type `Literal[42]`.

Anyway, this obviously shouldn't block merging; it's a pretty minor issue in the scheme of things.